### ISSUE TITLE
"tput: No value for $TERM and no -T specified" appears during test

### ISSUE THREAD

**Issue by KEINOS** [KEINOS/TPL-PHP-HelloWorld#4]
AFAIK, as of the template v1.0.6 2479287634cbb26a2708c03c237bf880183b5ac5 from composer, the following message appears during testing.

> `tput: No value for $TERM and no -T specified`

## How To Produce

```shellsession
$ composer create-project keinos/hello-world-tpl sample
...
$ cd $_
```
```shellsession
$ composer test
> ./tests/run-tests.sh
--------------------------------------------------------------------------------
  💡  Requirements all installed in local
--------------------------------------------------------------------------------
- RECOMMEND: Use "composer test local" command for faster test results.
--------------------------------------------------------------------------------
  Calling test container ...
--------------------------------------------------------------------------------

================================================================================
  Running Tests in Container
================================================================================
tput: No value for $TERM and no -T specified
--------------------------------------------------------------------------------
  💡  You are running inside Docker container.
--------------------------------------------------------------------------------

================================================================================
  Running tests
================================================================================
...
```

It happens if the tests're called via Docker.

```shellsession
$ docker run --rm test/local:7.4.3
tput: No value for $TERM and no -T specified
--------------------------------------------------------------------------------
  💡  You are running inside Docker container.
--------------------------------------------------------------------------------

================================================================================
  Running tests
================================================================================
```

If the `local` option is specified then it won't appear.

```shellsession
$ composer test local
> ./tests/run-tests.sh 'local'

================================================================================
  Running tests
================================================================================
```

Also, running the tests inside a container won't show the msg.

```shellsession
$ docker run --rm -it --entrypoint /bin/bash test/local:7.4.3
bash-5.0# ./tests/run-tests.sh
--------------------------------------------------------------------------------------------------
  💡  You are running inside Docker container.
--------------------------------------------------------------------------------------------------

=======================================================================
  Running tests
=======================================================================
```

### tl;dr

> Due to the `TERM` variable been set as "`dumb`", both `tput` command and `$TERM` variable are available. But, since it works as "[dumb terminal](https://en.wikipedia.org/wiki/Computer_terminal#Dumb_terminals)" no terminal width can be got by `cols`.
> This happens when the script was called via container's `entrypoint`.

### ts;dr

The message was caused in LINE 142 after calling the `run-test.sh` in a container recursively via docker-compose.

https://github.com/KEINOS/TPL-PHP-HelloWorld/blob/2479287634cbb26a2708c03c237bf880183b5ac5/tests/run-tests.sh#L139-L145

I inserted some lines for debugging, like below. 

```bash
which tput 2>/dev/null 1>/dev/null && {
    [ "${TERM:+unknown}" ] && {
        man tput
        echo "TERM: $TERM"
        tput cols
        SCREEN_WIDTH=$(tput cols);
    }
}
SCREEN_WIDTH=${SCREEN_WIDTH:-80};
```

Then I found out that `$TERM` value becomes "`dumb`" when the script was called via container. Running the script locally or inside the container `$TERM` value becomes `xterm-256color` or `xterm`.

## Suggested Fix

```diff
which tput 2>/dev/null 1>/dev/null && {
-    [ "${TERM:+unknown}" ] && {
+    [ "${TERM:+unknown}" ] && [ "$TERM" != "dumb" ] && {
        SCREEN_WIDTH=$(tput cols);
    }
}
SCREEN_WIDTH=${SCREEN_WIDTH:-80};
```

## REF.

```
Usage: tput [options] [command]

Options:
  -S <<       read commands from standard input
  -T TERM     use this instead of $TERM
  -V          print curses-version
  -x          do not try to clear scrollback

Commands:
  clear       clear the screen
  init        initialize the terminal
  reset       reinitialize the terminal
  capname     unlike clear/init/reset, print value for capability "capname"
```

- From `man tput`:

  > **-T**type indicates the type of terminal.  Normally this option is unnecessary, because the default is taken from the environment variable TERM.  If -T is specified, then the shell variables LINES and COLUMNS will be ignored, and the operating system will not be queried for the actual screen size.

- [Comment](https://unix.stackexchange.com/a/191538/377806) | How and where is $TERM interpreted? | Unix @ StackExchange

- About Dumb Terminal

  > In this context dumb terminals are sometimes dubbed glass Teletypes, for they essentially have the same limited functionality as does a mechanical Teletype. This type of dumb terminal is still supported on modern Unix-like systems by setting the environment variable TERM to dumb. Smart or intelligent terminals are those that also have the ability to process escape sequences, in particular the VT52, VT100 or ANSI escape sequences.
 
  <sup>（From: [Dumb terminals](https://en.wikipedia.org/wiki/Computer_terminal#Dumb_terminals) | Computer Terminal @ Wikipedia EN）</sup>

  > つまり、ダム端末はテレタイプ端末と同程度のことしかできない。Unix系システムではこのようなダム端末が今でもサポートされており、環境変数 TERM を dumb と設定することでダム端末と認識する。「インテリジェント端末」は、この文脈ではダム端末でないテキスト端末を意味する。

  <sup>（From: [ダム端末](https://ja.wikipedia.org/wiki/%E7%AB%AF%E6%9C%AB#%E3%83%80%E3%83%A0%E7%AB%AF%E6%9C%AB) | 端末 @ Wikipedia JA）</sup>

- [ダム端末](https://ja.wikipedia.org/wiki/%E3%83%80%E3%83%A0%E7%AB%AF%E6%9C%AB) @ Wikipedia JA

**Comment by KEINOS** [comment]
Came up with a more simple way.

```
$(tput cols 2>/dev/null 1>/dev/null) && {
    SCREEN_WIDTH=$(tput cols);
}
SCREEN_WIDTH=${SCREEN_WIDTH:-80};
```
